caom2utils/caom2utils/cadc_client_wrapper.py

```diff
@@ -0,0 +1,367 @@
+#  of the National Research             le nom du Conseil National de
+#  Council of Canada nor the            Recherches du Canada ni les noms
+#  names of its contributors may        de ses  participants ne peuvent
+#  be used to endorse or promote        être utilisés pour approuver ou
+#  products derived from this           promouvoir les produits dérivés
+#  software without specific prior      de ce logiciel sans autorisation
+#  written permission.                  préalable et particulière
+#                                       par écrit.
+#
+#  This file is part of the             Ce fichier fait partie du projet
+#  OpenCADC project.                    OpenCADC.
+#
+#  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+#  you can redistribute it and/or       vous pouvez le redistribuer ou le
+#  modify it under the terms of         modifier suivant les termes de
+#  the GNU Affero General Public        la “GNU Affero General Public
+#  License as published by the          License” telle que publiée
+#  Free Software Foundation,            par la Free Software Foundation
+#  either version 3 of the              : soit la version 3 de cette
+#  License, or (at your option)         licence, soit (à votre gré)
+#  any later version.                   toute version ultérieure.
+#
+#  OpenCADC is distributed in the       OpenCADC est distribué
+#  hope that it will be useful,         dans l’espoir qu’il vous
+#  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+#  without even the implied             GARANTIE : sans même la garantie
+#  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+#  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+#  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+#  General Public License for           Générale Publique GNU Affero
+#  more details.                        pour plus de détails.
+#
+#  You should have received             Vous devriez avoir reçu une
+#  a copy of the GNU Affero             copie de la Licence Générale
+#  General Public License along         Publique GNU Affero avec
+#  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+#  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+#                                       <http://www.gnu.org/licenses/>.
+#
+#  : 4 $
+#
+# ***********************************************************************
+#
+
+import logging
+import traceback
+from datetime import datetime, timezone
+from hashlib import md5
+from io import BytesIO
+from os import chdir, getcwd, path, stat
+
+from astropy.io import fits
+from urllib.parse import urlparse
+from cadcdata import FileInfo, CadcDataClient, StorageInventoryClient
+from cadcutils import exceptions
+
+
+__all__ = [
+    'get_file_type',
+    'get_local_file_headers',
+    'get_local_file_info',
+    'get_local_headers_from_fits',
+    'make_headers_from_string',
+    'StorageClientWrapper',
+]
```

Comment:
Judging by these functions, I would probably call this module `data` or similar. At the end it doesn't matter as it's imported in the `__init__.py` of the package so that the module name is optional in the imports.

Reply:
I called it `data_util` as I do not want any confusion with the soon-to-be-deprecated `data` service.